gobblin-compaction/src/main/java/org/apache/gobblin/compaction/verify/CompactionTimeRangeVerifier.java

```diff
@@ -73,10 +75,28 @@ public Result verify (FileSystemDataset dataset) {
       Period minTimeAgo = formatter.parsePeriod(minTimeAgoStr);
       latest = compactionStartTime.minus(minTimeAgo);
 
+      // get latest last run start time, we want to limit the duration between two compaction for the same dataset
+      if (state.contains(TimeBasedSubDirDatasetsFinder.MIN_RECOMPACTION_DURATION)) {
+        String minDurationStrList = this.state.getProp(TimeBasedSubDirDatasetsFinder.MIN_RECOMPACTION_DURATION);
+        String minDurationStr = getMachedLookbackTime(datasetName, minDurationStrList, TimeBasedSubDirDatasetsFinder.DEFAULT_MIN_RECOMPACTION_DURATION);
+        Period minDurationTime = formatter.parsePeriod(minDurationStr);
+        DateTime latestLastRunTime = compactionStartTime.minus(minDurationTime);
+        InputRecordCountHelper helper = new InputRecordCountHelper(state);
+        State compactState = helper.loadState(new Path(result.getDstAbsoluteDir()));
+        if (compactState.contains(CompactionSlaEventHelper.LAST_RUN_START_TIME)
+            && compactState.getPropAsLong(CompactionSlaEventHelper.LAST_RUN_START_TIME) > latestLastRunTime.getMillis()) {
+          log.warn("Last compaction for {} is {}, not before {}", dataset.datasetRoot(), new DateTime(compactState.getPropAsLong(CompactionSlaEventHelper.LAST_RUN_START_TIME), timeZone), latestLastRunTime);
+          return new Result(false, "Last compaction for " + dataset.datasetRoot() + " is not before" + latestLastRunTime);
+        }
+
+      }
+
       if (earliest.isBefore(folderTime) && latest.isAfter(folderTime)) {
         log.debug("{} falls in the user defined time range", dataset.datasetRoot());
         return new Result(true, "");
       }
+    } catch (RuntimeException e) {
```

Comment:
Why are we catching an unchecked exception and just throw it without doing anything here?

Reply:
It's due to the findBugsMain, seems we catch exception and not throw it intentionally, I will change the rule to pass it.